src/io/flutter/logging/FlutterConsoleLogManager.java

```diff
@@ -126,95 +127,157 @@ public void handleFlutterErrorEvent(@NotNull Event event) {
-      if (lastLevel != null && lastLevel != property.getLevel()) {
-        if (lastLevel == DiagnosticLevel.hint || property.getLevel() == DiagnosticLevel.hint) {
-          console.print("\n", NORMAL_CONTENT_TYPE);
+    if (terseError) {
+      for (DiagnosticsNode property : diagnosticsNode.getInlineProperties()) {
+        printTerseNodeProperty(console, "", property);
+      }
+    }
+    else {
+      DiagnosticLevel lastLevel = null;
+
+      for (DiagnosticsNode property : diagnosticsNode.getInlineProperties()) {
+        // Add blank line between hint and non-hint properties.
+        if (lastLevel != null && lastLevel != property.getLevel()) {
+          if (lastLevel == DiagnosticLevel.hint || property.getLevel() == DiagnosticLevel.hint) {
+            console.print("\n", NORMAL_CONTENT_TYPE);
+          }
         }
+
+        lastLevel = property.getLevel();
+
+        printDiagnosticsNodeProperty(console, "", property, null, false);
       }
+    }
 
-      lastLevel = property.getLevel();
+    console.print(StringUtil.repeat(errorSeparatorChar, errorSeparatorLength) + "\n\n", TITLE_CONTENT_TYPE);
+  }
 
-      printDiagnosticsNodeProperty(console, "", property, null, false);
+  private void printTerseNodeProperty(ConsoleView console, String indent, DiagnosticsNode property) {
+    boolean skip = true;
+
+    if (property.getLevel() == DiagnosticLevel.summary) {
+      skip = false;
+    }
+    else if (property.hasChildren()) {
+      try {
+        final CompletableFuture<ArrayList<DiagnosticsNode>> future = property.getChildren();
+        final ArrayList<DiagnosticsNode> children = future.get();
+        if (children.stream().noneMatch(DiagnosticsNode::hasChildren)) {
+          skip = false;
+        }
+      }
+      catch (InterruptedException | ExecutionException e) {
+        LOG.warn(e);
+      }
+    }
+
+    if (skip) {
+      return;
+    }
+
+    final ConsoleViewContentType contentType = getContentTypeFor(property.getLevel());
+
+    console.print(indent, contentType);
+
+    if (property.getShowName()) {
+      console.print(property.getName(), contentType);
+
+      if (property.getShowSeparator()) {
+        console.print(property.getSeparator() + " ", contentType);
+      }
     }
 
-    console.print(errorSeparator + "\n\n", TITLE_CONTENT_TYPE);
+    final String description = property.getDescription() == null ? "" : property.getDescription();
+    console.print(description + "\n", contentType);
+
+    if (property.hasChildren()) {
+      try {
+        final CompletableFuture<ArrayList<DiagnosticsNode>> future = property.getChildren();
+        final ArrayList<DiagnosticsNode> children = future.get();
+
+        final String childIndent = getChildIndent(indent, property);
+        for (DiagnosticsNode child : children) {
+          printDiagnosticsNodeProperty(console, childIndent, child, contentType, false);
+        }
+
+        if (property.hasInlineProperties()) {
```

Comment:
by convention properties should always be displayed before children instead of after children.

Reply:
OK, I'll re-order this (and post new screen shots if things change).

Reply:
Re-ordering did end up eliminating a bunch of duplicate code.